airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_client.py

```diff
@@ -0,0 +1,111 @@
+        refresh_token=SP_CREDENTIALS["refresh_token"], lwa_app_id=SP_CREDENTIALS["lwa_app_id"],
+        lwa_client_secret=SP_CREDENTIALS["lwa_client_secret"], aws_secret_key=SP_CREDENTIALS["aws_secret_key"],
+        aws_access_key=SP_CREDENTIALS["aws_access_key"], marketplace=MARKETPLACE)
+
+    assert ORDERS_RESPONSE == base_client.check_connection()
+
+
+def test_get_cursor_state():
+    cursor_value = "2021-05-02"
+    end_date = "2021-06-02"
+
+    base_client = BaseClient(
+        role_arn=SP_CREDENTIALS["role_arn"], start_date=SP_CREDENTIALS["start_date"],
+        refresh_token=SP_CREDENTIALS["refresh_token"], lwa_app_id=SP_CREDENTIALS["lwa_app_id"],
+        lwa_client_secret=SP_CREDENTIALS["lwa_client_secret"], aws_secret_key=SP_CREDENTIALS["aws_secret_key"],
+        aws_access_key=SP_CREDENTIALS["aws_access_key"], marketplace=MARKETPLACE)
+
+    assert "2021-05-03" == base_client._get_cursor_state(
+        cursor_value=cursor_value, end_date=end_date)
+
+    cursor_value = "2021-02-28"
+    end_date = "2021-03-02"
+
+    assert "2021-03-03" == base_client._get_cursor_state(
+        cursor_value=cursor_value, end_date=end_date)
+
+
+def test_get_records():
+    data = {"document": "name\ttest\nairbyte\t1"}
+    base_client = BaseClient(
+        role_arn=SP_CREDENTIALS["role_arn"], start_date=SP_CREDENTIALS["start_date"],
+        refresh_token=SP_CREDENTIALS["refresh_token"], lwa_app_id=SP_CREDENTIALS["lwa_app_id"],
+        lwa_client_secret=SP_CREDENTIALS["lwa_client_secret"], aws_secret_key=SP_CREDENTIALS["aws_secret_key"],
+        aws_access_key=SP_CREDENTIALS["aws_access_key"], marketplace=MARKETPLACE)
+
+    assert [{"name": "airbyte", "test": "1"}] == base_client._get_records(data)
+
+
+def test_apply_conversion_window():
+    current_date = "2021-03-04"
+    base_client = BaseClient(
+        role_arn=SP_CREDENTIALS["role_arn"], start_date=SP_CREDENTIALS["start_date"],
+        refresh_token=SP_CREDENTIALS["refresh_token"], lwa_app_id=SP_CREDENTIALS["lwa_app_id"],
+        lwa_client_secret=SP_CREDENTIALS["lwa_client_secret"], aws_secret_key=SP_CREDENTIALS["aws_secret_key"],
+        aws_access_key=SP_CREDENTIALS["aws_access_key"], marketplace=MARKETPLACE)
+
+    assert "2021-02-18" == base_client._apply_conversion_window(current_date)
+
+
+def test_convert_array_into_dict():
+    headers = ["name", "test"]
+    records = ["airbyte\t1"]
+    assert [{"name": "airbyte", "test": "1"}
+            ] == BaseClient._convert_array_into_dict(headers, records)
+
+
+def test_increase_date_by_month():
+    current_date = "2021-03-04"
+    assert "2021-04-04" == BaseClient._increase_date_by_month(current_date)
+
+
+def test_get_date_parameters():
+    current_date = "2021-03-04"
+    assert ("2021-03-04", "2021-04-04") == BaseClient._get_date_parameters(current_date)
+    current_date = "2021-05-24"
+    assert ("2021-05-24", "2021-05-30") == BaseClient._get_date_parameters(current_date)
```

Comment:
will this test fail once the date advances? if so what if we instantiate the current_date using dynamic functions for example: 

```
def fmt_date(date):
    return datetime.strftime(date, '%Y-%m-%d')

def test_get_date_parameters():
  # If the start date is more than one month ago then we expect a full 30 day increment
  now = datetime.now()
  two_months_ago = fmt_date(now - relativedelta(months=2))
  one_month_ago = fmt_date(now - relativedelta(months=1))
  assert (two_months_ago, one_month_ago) == BaseClient._get_date_parameters(two_months_ago)
  
  # If the start date is less than one month ago we expect to advance no later than today
  one_week_ago = fmt_date(now - relativedelta(weeks=1))
  yesterday = fmt_date(now - relativedelta(days=1))
  assert (one_week_ago, yesterday) == BaseClient._get_date_parameters(one_week_ago)
```